src/routes/patreon.ts

```diff
@@ -0,0 +1,81 @@
+import express from 'express';
+import { patreonAgent } from '../clients/patreon/patreonAgent';
+
+const router = express.Router();
+
+router.get('/oauth/redirect', async (req, res) => {
+  console.log('Start: Link to Patreon...');
+  const { code, state } = req.query;
+
+  // TODO-kev: Consider what happens if someone clicks deny. Rework error msg
+  if (!code) {
+    return res.redirect(
+      // @ts-ignore
+      `${req.session.patreonOriginalUrl}?error=You have denied access`,
+    );
+  }
+
+  // @ts-ignore
+  if (state !== req.session.patreonAuthState) {
```

Comment:
What's this state for? Can we remove `req.session.patreonAuthState`?

Reply:
Prevent against CSRF attacks